exercises/k9-data-structures/lists/linked-list-vs-array/array/problem.js

```diff
@@ -9,9 +9,37 @@ export class MyArray {
         return this;
     }
 
-    size() {}
-    prepend(index) {}
-    get(index) {}
-    insertAt(index, value) {}
-    forEach(callback) {}
+    size() {
+        return this.length
+    }
+    prepend(value) {
+        const temp = {};
+        temp[0] = value;
+        for (let i = 0; i < this.length; i++) {
+            temp[i + 1] = this.nodes[i]
+        }
+        this.nodes = temp;
+        this.length++
+        return this
+    }
+    get(index) {
+        if (index < this.length) {
+            return this.nodes[index];
+        }
+        return null;
+    }
+    insertAt(index, value) {
+        const temp = { ...this.nodes }
+        this.nodes[index] = value
+        for (let i = index + 1; i < this.length + 1; i++) {
```

Comment:
For extra points, there is a way to make this loop shorter if the index is in the middle of the array 😉 Consider inserting `"A"` into `[1,2,3,4,5]` to get `[1,2,3,"A",4,5]`... which indexes do we _need_ to update?

Reply:
Just the final 2?

Reply:
my bad